editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/wizards/AddGlobalBeanWizard.java

```diff
@@ -0,0 +1,135 @@
+	private CamelFile camelFile;
+	private ComponentManager componentManager;
+
+	public AddGlobalBeanWizard(CamelFile camelFile, ComponentModel componentModel) {
+		super();
+		this.camelFile = camelFile;
+		this.dbc = new DataBindingContext();
+		this.componentManager = new ComponentManager(componentModel);
+		setWindowTitle("Add Bean");
+		setNeedsProgressMonitor(true);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.jface.wizard.Wizard#addPages()
+	 */
+	@Override
+	public void addPages() {
+		super.addPages();
+		globalBeanPage = new GlobalBeanWizardPage(dbc, "Bean Definition",
+				"Specify details for the new bean definition.", camelFile);
+		addPage(globalBeanPage);
+
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.fusesource.ide.camel.editor.provider.ext.
+	 * GlobalConfigurationTypeWizard#getGlobalConfigrationElementNode()
+	 */
+	@Override
+	public Element getGlobalConfigurationElementNode() {
+		return globalConfigurationNode;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.fusesource.ide.camel.editor.provider.ext.
+	 * GlobalConfigurationTypeWizard#setGlobalConfigrationElementNode(org.w3c.
+	 * dom.Element)
+	 */
+	@Override
+	public void setGlobalConfigurationElementNode(Element node) {
+		this.globalConfigurationNode = node;
+	}
+
+	/**
+	 * @return the component
+	 */
+	public Component getComponent() {
+		Object component = componentManager.getComponentForTag(CamelBean.BEAN_NODE);
+		if (component != null && component instanceof Component) {
+			return (Component) component;
+		}
+		return null;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.jface.wizard.Wizard#performFinish()
+	 */
+	@Override
+	public boolean performFinish() {
+		final String prefixNS = camelFile.getRouteContainer().getXmlNode().getPrefix();
+		globalConfigurationNode = camelFile.createElement(CamelBean.BEAN_NODE, prefixNS);
+		globalConfigurationNode.setAttribute(CamelBean.PROP_ID, globalBeanPage.getId());
+		if (!Strings.isBlank(globalBeanPage.getClassname())) {
+			globalConfigurationNode.setAttribute(CamelBean.PROP_CLASS, globalBeanPage.getClassname());
+		}
+		if (!globalBeanPage.getArgumentsList().isEmpty()) {
+			for (AbstractCamelModelElement argument : globalBeanPage.getArgumentsList()) {
+				Element children = (Element) globalConfigurationNode.getChildNodes();
+				children.appendChild(argument.getXmlNode());
+			}
+		}
+		if (!globalBeanPage.getPropertyList().isEmpty()) {
+			for (AbstractCamelModelElement property : globalBeanPage.getPropertyList()) {
+				Element children = (Element) globalConfigurationNode.getChildNodes();
+				children.appendChild(property.getXmlNode());
+			}
+		}
+		setGlobalConfigurationElementNode(globalConfigurationNode);
+		return true;
+	}
+
+	/*
+	 * Though this seems nonsensical at this point, it somehow resets the wizard to the point
+	 * where the updated XML element is updated correctly in the model. 
+	 */
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
```

Comment:
Note the comment above this line. I tried removing them, but was right back to things not being persisted properly in the underlying model. So I'm open to suggestions. :) Is there a way for me to safely ignore this comment in this case?